plugins/webhooks/index.js

```diff
@@ -4,9 +4,40 @@ const boom = require('boom');
 const joi = require('joi');
 const winston = require('winston');
 const workflowParser = require('screwdriver-workflow-parser');
-
+const CHECKOUT_URL_SCHEMA = require('screwdriver-data-schema').config.regex.CHECKOUT_URL;
+const CHECKOUT_URL_SCHEMA_REGEXP = new RegExp(CHECKOUT_URL_SCHEMA);
 const WAIT_FOR_CHANGEDFILES = 1.8;
 
+/**
+ * Determine "startFrom" with type, action and branches
+ * @param {String} action          SCM webhook action type
+ * @param {String} type            Triggered SCM event type ('pr' or 'repo')
+ * @param {String} targetBranch    The branch against which commit is pushed
+ * @param {String} pipelineBranch  The pipeline branch
+ * @returns {String}               startFrom
+ */
+function determineStartFrom(action, type, targetBranch, pipelineBranch) {
```

Comment:
We should write unit test for this function. Its test will be useful in the second pass implementation.

Reply:
added!